src/ObjCRuntime/Runtime.cs

```diff
@@ -2262,6 +2380,30 @@ public int GetHashCode (Type? obj)
 		}
 	}
 
+	internal class StringEqualityComparer : IEqualityComparer<string> {
+		public bool Equals (string? x, string? y)
+		{
+			return string.Equals (x, y, StringComparison.Ordinal);
+		}
+		public int GetHashCode (string? obj)
+		{
+			if (obj is null)
```

Comment:
same 
`return obj is null ? 0 : obj.GetHashCode ();`

Reply:
This could even be simpler:

```cs
return obj?.GetHashCode () ?? 0;
```

Reply:
Fixed!